tests/Unit/ArraysTest.php

```diff
@@ -61,6 +61,7 @@ public function testMapAssoc(): void
 
         static::assertSame([], Arrays::mapAssoc([], $callback));
         static::assertSame(['foo' => 'bar'], Arrays::mapAssoc([['foo', 'bar']], $callback));
+        static::assertSame([2 => false, 4 => true, 6 => false], Arrays::mapAssoc([1, 2, 3], static fn(int $val) => [$val * 2, $val % 2 === 0]));
```

Comment:
The `int` is unfortunately required in `static fn(int $val)` the callback. The error message from phpstan:

` 64     Unable to resolve the template type K in call to method static method DR\Utils\Arrays::mapAssoc()`

I tried multiple things, but it's either casting $val to int of the first array value, or typing the callback argument.